### PR TITLE
Fix iteration over invalid slots

### PR DESCRIPTION
**Pull request by D-Cysteine**

### CODE REVIEW

File: src/main/java/gregtech/common/tileentities/automation/GT_MetaTileEntity_ChestBuffer.java [hunk 1]
```diff
@@ -125,9 +125,9 @@ public int compare(ItemStack o1, ItemStack o2) {
     protected void fillStacksIntoFirstSlots() {
         sortStacks();
         // Merge small stacks together
-        for (int i = 0; i < this.mInventory.length-1;) {
+        for (int i = 0; i < this.mInventory.length - 2;) {
```
Comment (Glease) [inline src/main/java/gregtech/common/tileentities/automation/GT_MetaTileEntity_ChestBuffer.java:128]: Maybe add a comment to explain why it's -2?

File: src/main/java/gregtech/common/tileentities/automation/GT_MetaTileEntity_ItemDistributor.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
 
     @Override
     protected void fillStacksIntoFirstSlots() {
-        for (int i = 0; i < this.mInventory.length - 1; i++) {
+        for (int i = 0; i < this.mInventory.length - 2; i++) {
```
Comment (Glease) [inline src/main/java/gregtech/common/tileentities/automation/GT_MetaTileEntity_ItemDistributor.java:54]: Same here

File: src/main/java/gregtech/common/tileentities/automation/GT_MetaTileEntity_ChestBuffer.java (hunk 1 above)
Comment (D-Cysteine) [reply src/main/java/gregtech/common/tileentities/automation/GT_MetaTileEntity_ChestBuffer.java:130]: Added

File: src/main/java/gregtech/common/tileentities/automation/GT_MetaTileEntity_ItemDistributor.java (hunk 2 above)
Comment (D-Cysteine) [reply src/main/java/gregtech/common/tileentities/automation/GT_MetaTileEntity_ItemDistributor.java:56]: Added

**Comment by D-Cysteine** [comment]
FYI I have an idea to fix this more properly:
https://github.com/D-Cysteine/GT5-Unofficial/commit/71af9df9bfe7185413fe1fb7253bc3d14eb0f3a3
Need to check that this doesn't break anything though as it's a bit more extensive

**Comment by Dream-Master** [comment]
@D-Cysteine let us know when its done.

**Comment by D-Cysteine** [comment]
I probably won't get to fully testing it for a while. It shouldn't block this PR anyway; it's meant to build on top of this one.
If you're feeling brave, or if others have time to help with testing it - it's complete, and might be fully working already.

**Comment by Dream-Master** [comment]
@D-Cysteine so we merge this and add a second pr later ?

**Comment by Glease** [comment]
Gotta pause a day or two. Power outage has damaged the file system. Currently running chkdsk through. Let's hope not too much stuff is lost...

________________________________

**Comment by D-Cysteine** [comment]
> @D-Cysteine so we merge this and add a second pr later ?

Yup, that was my plan

> Gotta pause a day or two. Power outage has damaged the file system. Currently running chkdsk through. Let's hope not too much stuff is lost...

=( Hope everything is okay

**Review by Glease — APPROVED** [review]

**Pull request merged**